### ISSUE TITLE
Does FlinkPulsarSource support multiple thread to consume one topic? 

### ISSUE THREAD

**Issue by Johnnyhooyo** [streamnative/pulsar-flink#356]
now it's seems to only single thread for each topic even partitioned topic.

**Comment by jianyun8023** [comment]
@Johnnyhooyo  Yes, if you need faster consumption, you can try to increase the parallelism of flink job.

**Comment by Johnnyhooyo** [comment]
> @Johnnyhooyo Yes, if you need faster consumption, you can try to increase the parallelism of flink job.

In my job, the topic has 2 partition, and parallelism of source is 4.  then just 2 thread consume the message.

I see the connector use reader to consume message,  and the reader not support "subscriptionType.shared"

**Comment by Johnnyhooyo** [comment]
![image](https://user-images.githubusercontent.com/35983936/122342171-ddd8de00-cf76-11eb-8ff3-5e551c22c56e.png)
look at that, the parallelism of job is 4 @jianyun8023

**Comment by shulaoh** [comment]
hi , I hit same problem. I set the pulsar partition to 1, enable-key-hash-range to true, and use flink source with 4 threads to read it, only one thread can consume message.

seems shared read doesn't take effect!

**Comment by shulaoh** [comment]
just curious why this issue is not responded since it is a fundamental feature...

**Comment by jianyun8023** [comment]
@Johnnyhooyo @shulaoh Sorry, the number of Pulsar topic partitions and Flink parallelism is 1:1. This is because the connector uses the Pulsar Reader API to consume data, and the Reader uses exclusive mode to consume, so only one reader can consume a partition.

If you want to use Pulsar consume API to solve this limitation, you can read about [Pulsar Source in Flink 1.14](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/connectors/datastream/pulsar/).
